### DISCUSSION TITLE
Is it possible to override the Content-Disposition header for a webhook upload?

### DISCUSSION THREAD

**Discussion by gstark** [gotenberg/gotenberg#494] [Q&A]
Scenario: Using webhook to upload generated PDF to a pre-signed AWS S3 URL.

The webhook middleware provides a `Content-Disposition` header. However, when I specify one via `Gotenberg-Webhook-Extra-Http-Headers` it is not considered. The webhook client processes the extra headers first and then internally generated headers, thus it appears not to be possible to override Gotenberg's content disposition headers.

See:

https://github.com/gotenberg/gotenberg/blob/d4c47cecf2ca1c35dc504073ce818862aa1bb180/pkg/modules/webhook/client.go#L46-L49

and

https://github.com/gotenberg/gotenberg/blob/d4c47cecf2ca1c35dc504073ce818862aa1bb180/pkg/modules/webhook/client.go#L72-L74

I can understand the rationale for preventing `Gotenberg-Webhook-Extra-Http-Headers` to override the client's headers. However, this precludes specifying a `Content-Disposition` when uploading to a service like `AWS S3`

**Comment by gulien** [comment]
Hello @gstark,

I’ve created the issue #495 to discuss your problem 😊
